packages/typespec-ts/test/integration/encode.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { assert } from "chai";
+import EncodeDurationClientFactory, {
+  EncodeDurationClient
+} from "./generated/encode/src/index.js";
+import { matrix } from "../util/matrix.js";
+describe("EncodeDurationClient Rest Client", () => {
+  let client: EncodeDurationClient;
+
+  beforeEach(() => {
+    client = EncodeDurationClientFactory({
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  interface TypeDetail {
+    type: "default" | "float-seconds" | "int32-seconds" | "iso8601";
+    defaultValue: any;
+  }
+
+  const durationData: TypeDetail[] = [
+    {
+      type: "default",
+      defaultValue: "P40D"
+    },
+    {
+      type: "float-seconds",
+      defaultValue: 35.621
+    },
+    {
+      type: "int32-seconds",
+      defaultValue: 36
+    },
+    {
+      type: "iso8601",
+      defaultValue: "P40D"
+    }
+  ];
+
+  matrix([durationData], async (params: TypeDetail) => {
```

Comment:
Personally, I don't like using matrix and use as any to do the type casting to work around the compiler error. though it looks more succinct. @joheredi how do you think ?

Reply:
Generally I would leverage the first generation code to ensure the type info is expected and later using any change if happened to check type changes.  I agree this is not perfect but easy to extend.